packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2

```diff
@@ -457,152 +550,185 @@ class Model(_MyMutableMapping):
+    try:
+        if sys.version_info >= (3, 8):
+            from typing import (
+                Literal,
+            )  # pylint: disable=no-name-in-module, ungrouped-imports
+        else:
+            from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
 
-        # is it a literal?
-        try:
-            if sys.version_info >= (3, 8):
-                from typing import Literal  # pylint: disable=no-name-in-module, ungrouped-imports
-            else:
-                from typing_extensions import Literal  # type: ignore  # pylint: disable=ungrouped-imports
+        if annotation.__origin__ == Literal:
+            return None
+    except AttributeError:
+        pass
 
-            if annotation.__origin__ == Literal:
-                return None
-        except AttributeError:
-            pass
+    if getattr(annotation, "__origin__", None) is typing.Union:
 
-        if getattr(annotation, "__origin__", None) is typing.Union:
-            def _deserialize_with_union(union_annotation, obj):
-                for t in union_annotation.__args__:
-                    try:
-                        return _deserialize(t, obj, module)
-                    except DeserializationError:
-                        pass
-                raise DeserializationError()
-            return functools.partial(_deserialize_with_union, annotation)
-
-        # is it optional?
-        try:
-            # right now, assuming we don't have unions, since we're getting rid of the only
-            # union we used to have in msrest models, which was union of str and enum
-            if any(a for a in annotation.__args__ if a == type(None)):
+        def _deserialize_with_union(union_annotation, obj):
+            for t in union_annotation.__args__:
+                try:
+                    return _deserialize(t, obj, module, rf)
+                except DeserializationError:
+                    pass
+            raise DeserializationError()
 
-                if_obj_deserializer = _get_deserialize_callable_from_annotation(
-                    next(a for a in annotation.__args__ if a != type(None)), module, rf
-                )
+        return functools.partial(_deserialize_with_union, annotation)
 
-                def _deserialize_with_optional(if_obj_deserializer: typing.Optional[typing.Callable], obj):
-                    if obj is None:
-                        return obj
-                    return _deserialize_with_callable(if_obj_deserializer, obj)
+    # is it optional?
+    try:
+        # right now, assuming we don't have unions, since we're getting rid of the only
+        # union we used to have in msrest models, which was union of str and enum
+        if any(a for a in annotation.__args__ if a == type(None)):
+            if_obj_deserializer = _get_deserialize_callable_from_annotation(
+                next(a for a in annotation.__args__ if a != type(None)), module, rf
+            )
```

Comment:
It is a little chaotic when format and new added code exist at the same time. Next time it is better to make a split PR to commit format part.